src/logic.js

```diff
@@ -501,7 +501,42 @@ define([
                 .map(function(ref) {
                     return [ref.path, null];
                 }).object().value();
-        }
+        },
+        /*
+         * Returns object describing references used in the logic manager
+         *
+         * {
+         *   /path/of/referenced/mug: {
+         *     /path/of/mug/referencing: property where ref occurs
+         *   }
+         * }
+         *
+         * Optionally can filter based on hashtag path
+         */
+        findUsages: function (path) {
+            var _this = this,
+                form = _this.form,
+                tableData = {},
+                refs = _this.reverse;
+
+            _.each(refs, function (value, key) {
+                var usedMug = form.getMugByUFID(key),
+                    mugReferences = {};
+                if (path && path !== usedMug.hashtagPath) {
+                    return;
+                }
+                _.each(value, function (value, key) {
+                    _.each(value, function (value) {
```

Comment:
slightly better naming although I think it will still take me a bit to figure out what's going on when I come back to it https://github.com/dimagi/Vellum/pull/794/commits/e476b06bea4390dd0bbbf13b901ef20c6aec3108